### PR TITLE
docs: README.md updated to be in sync with docs.minio.io

### PR DESCRIPTION
**Pull request by koolhead17**
Added API Reference Section.
Added individual links to each API.
Brought it as close as possible to docs site.
Add constructor so that this document is complete in itself.
Removed How to Run this example section.
Changed desc to description in the table heading.
Call it Full Docs Site to Complete Documentation.

### CODE REVIEW

**Comment by harshavardhana** [comment]
// cc @GarimaKapoor - need your review.

**Comment by GarimaKapoor** [comment]
Remove the block, its repetitive from the file-uploader example.

```
from minio import Minio
from minio.error import ResponseError

minioClient = Minio('play.minio.io:9000',
access_key='Q3AM3UQ867SPQQA43P2F',
secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
secure=True)
```
  
otherwise all good!

**Comment by harshavardhana** [comment]
README.rst is necessary for python packaging, you should add it back.

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Minio Python Library for Amazon S3 Compatible Cloud Storage [![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/minio/minio?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (harshavardhana) [inline README.rst]: BTW this should reflect the same file as README.md

Comment (koolhead17) [inline README.rst]: Let me send a Patch which reflects the same in RST. 👍

**Comment by harshavardhana** [comment]
@GarimaKapoor is this clean?

**Comment by koolhead17** [comment]
@harshavardhana noticed one more minor issue. Let me send a patch fixing it.

**Comment by koolhead17** [comment]
@GarimaKapoor @harshavardhana need feedback on this so we can close this. 😄

**Comment by harshavardhana** [comment]
LGTM, Thanks :+1:

**Pull request merged**